packages/navigator/src/electron-browser/electron-navigator-menu-contribution.ts

```diff
@@ -0,0 +1,74 @@
+ * Copyright (C) 2021 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Command, CommandContribution, CommandRegistry, MenuContribution, MenuModelRegistry } from '@theia/core';
+import { CommonCommands, KeybindingContribution, KeybindingRegistry } from '@theia/core/lib/browser';
+import { WidgetManager } from '@theia/core/lib/browser/widget-manager';
+import * as electron from '@theia/core/shared/electron';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { FileStatNode } from '@theia/filesystem/lib/browser';
+import { FileNavigatorWidget, FILE_NAVIGATOR_ID } from '../browser';
+import { NavigatorContextMenu } from '../browser/navigator-contribution';
+import { isWindows, isOSX } from '@theia/core/lib/common/os';
+
+export const OPEN_CONTAINING_FOLDER = Command.toDefaultLocalizedCommand({
+    id: 'revealFileInOS',
+    category: CommonCommands.FILE_CATEGORY,
+    label: isWindows ? 'Reveal File in File Explorer' :
+        isOSX ? 'Reveal in Finder' :
+        /* linux */ 'Open Containing Folder'
+});
+
+@injectable()
+export class ElectronNavigatorMenuContribution implements MenuContribution, CommandContribution, KeybindingContribution {
+
+    @inject(WidgetManager)
+    protected readonly widgetManager: WidgetManager;
+
+    registerCommands(commands: CommandRegistry): void {
+        commands.registerCommand(OPEN_CONTAINING_FOLDER, {
+            isEnabled: () => this.getSelectedFileStatNodes().length > 0,
+            isVisible: () => this.getSelectedFileStatNodes().length > 0,
+            execute: () => {
+                this.getSelectedFileStatNodes().forEach(async node => {
+                    electron.shell.showItemInFolder(node.uri.path.toString());
+                });
+            }
+        });
+    }
+    registerMenus(menus: MenuModelRegistry): void {
+        menus.registerMenuAction(NavigatorContextMenu.NAVIGATION, {
+            commandId: OPEN_CONTAINING_FOLDER.id,
+            label: OPEN_CONTAINING_FOLDER.label
+        });
+    }
+
+    registerKeybindings(keybindings: KeybindingRegistry): void {
+        keybindings.registerKeybinding({
+            command: OPEN_CONTAINING_FOLDER.id,
+            keybinding: isWindows ? 'shift+alt+r' : 'alt+ctrlcmd+r'
```

Comment:
I believe this keybinding conflicts with `open recent workspace...`:

![image](https://user-images.githubusercontent.com/40359487/146392566-96eaf50d-59cd-477c-8b78-8a760366688e.png)

We should add a proper `when` clause of when the keybinding should be active, for example when a resource is selected in the navigator.

Reply:
I fixed the keybinding and added `when` clause in the updated commit.